addons/mail/models/mail_thread.py

```diff
@@ -2397,6 +2398,10 @@ def _notify_thread_by_email(self, message, recipients_data, msg_vals=False,
             return True
         force_send = self.env.context.get('mail_notify_force_send', force_send)
 
+        if scheduled_date:
```

Comment:
An issue with that model is that you create discuss notifications right now. They should also be created when we really notify...